derp/derp_client.go

```diff
@@ -369,6 +370,38 @@ type KeepAliveMessage struct{}
 
 func (KeepAliveMessage) msg() {}
 
+// HealthMessage is a one-way message from server to client, declaring the
+// connection health state.
+type HealthMessage struct {
+	// Problem, if non-empty, is a description of why the connection
+	// is unhealthy.
+	//
+	// The empty string means the connection is healthy again.
+	//
+	// The default condition is healthy, so the server doesn't
+	// broadcast a HealthMessage until a problem exists.
+	Problem string
```

Comment:
In programming we have a thousand ways to say something is wrong, but not one way to say everything is fine.